### PR TITLE
Swap `scalameta` `trees` for `semanticdb-shared` & bump `scalameta` to 4.9.8

### PR DESCRIPTION
**Pull request by Gedochao**
Supersedes #3016

### CODE REVIEW

**Comment by Gedochao** [comment]
https://github.com/jline/jline3/issues/954
Ok... so it seems we swallow Jline 3.25.1 along with `scalameta` 4.9.8 here.
That version of JLine seems to require Graal 23+...  so we need to bump Graal first somehow for our native images to get generated properly.
Otherwise we're getting:
```
Error: Could not find option 'UnlockExperimentalVMOptions' from 'jar:file:///data/cp/jline-3.25.1.jar!/META-INF/native-image/org.jline/jline-terminal/native-image.properties'. Use -H:PrintFlags= to list all available options.
Could not find option 'UnlockExperimentalVMOptions' from 'jar:file:///data/cp/jline-3.25.1.jar!/META-INF/native-image/org.jline/jline-terminal-jansi/native-image.properties'. Use -H:PrintFlags= to list all available options.
Could not find option 'UnlockExperimentalVMOptions' from 'jar:file:///data/cp/jline-3.25.1.jar!/META-INF/native-image/org.jline/jline-terminal-jni/native-image.properties'. Use -H:PrintFlags= to list all available options.
Could not find option 'UnlockExperimentalVMOptions' from 'jar:file:///data/cp/jline-3.25.1.jar!/META-INF/native-image/org.jline/jansi/native-image.properties'. Use -H:PrintFlags= to list all available options.
Could not find option 'UnlockExperimentalVMOptions' from 'jar:file:///data/cp/jline-3.25.1.jar!/META-INF/native-image/org.jline/jline-terminal-jna/native-image.properties'. Use -H:PrintFlags= to list all available options.
Could not find option 'UnlockExperimentalVMOptions' from 'jar:file:///data/cp/jline-3.25.1.jar!/META-INF/native-image/org.jline/jline-terminal-ffm/native-image.properties'. Use -H:PrintFlags= to list all available options.
Could not find option 'UnlockExperimentalVMOptions' from 'jar:file:///data/cp/jline-3.25.1.jar!/META-INF/native-image/org.jline/jline-native/native-image.properties'. Use -H:PrintFlags= to list all available options.
```
And bumping Graal is tricky, since the aarch64 platform has been dropped from CE after the version we're at.
I'll try looking into it separately, but it seems this is blocked for now.

File: project/deps.sc [hunk 1]
```diff
@@ -180,16 +180,15 @@ object Deps {
   def scalaJsLogging   = ivy"org.scala-js:scalajs-logging_2.13:1.1.1"
   // Force using of 2.13 - is there a better way?
   def scalaJsTestAdapter = ivy"org.scala-js:scalajs-sbt-test-adapter_2.13:${Scala.scalaJs}"
-  // Force using of 2.13 - is there a better way?
-  def scalametaTrees = ivy"org.scalameta:trees_2.13:${Versions.scalaMeta}"
```
Comment (tgodzik) [inline project/deps.sc:184]: So we never really needed trees actually? This actually seems like the wrong dependency was used

**Review by tgodzik — APPROVED** [review]
Thanks for digging into this!

File: project/deps.sc (hunk 1 above)
Comment (Gedochao) [reply project/deps.sc:184]: It seems we only needed it for the SemanticDB stuff, yeah.

**Pull request merged**